src/recipe/rust.rs

```diff
@@ -57,14 +57,20 @@ impl Rust {
             _ if has_cargo_config => "".to_string(),
             BuildEnv::Debug => format!("RUSTFLAGS=\"{}\"", BASE_RUSTFLAGS.to_string()),
             BuildEnv::Release => {
+                let mut args = String::from(format!("RUSTFLAGS=\"{} ", BASE_RUSTFLAGS));
                 if config.always_debug {
-                    format!(
-                        "RUSTFLAGS=\"{} {} {}\"",
-                        BASE_RUSTFLAGS, RELEASE_RUSTFLAGS, ALWAYS_DEBUG_RUSTFLAGS
-                    )
+                    args.push_str(&format!("{} ", ALWAYS_DEBUG_RUSTFLAGS));
+                }
+                if config.debug_info {
+                    // https://doc.rust-lang.org/rustc/codegen-options/index.html#debuginfo
+                    // The `-g` flag is an alias for -C debuginfo=2,
+                    // which will generate full debug info
+                    args.push_str("-g ");
                 } else {
-                    format!("RUSTFLAGS=\"{} {}\"", BASE_RUSTFLAGS, RELEASE_RUSTFLAGS)
+                    args.push_str(&format!("{} ", RELEASE_RUSTFLAGS));
```

Comment:
`RELEASE_RUSTFLAGS` is a very generic name, chances are more flags for release version might be included here. What we should do, is to introduce a new flag just for handling ELF stripping. When `debug_info` is enabled, only the flag for ELF stripping should be removed, there might well be other release version RUSTFLAGS that shall be used even when `debug_info` is enabled.